<launch>
    <arg name = "veh" doc = "duckiebot"/>
    <arg name = "pkg_name" value = "motion_planner"/>
    <arg name = "node_name" default = "control_node"/>
    <arg name = "parameter_file_name" default = "default"/>
    <group ns = "$(arg veh)">
        <remap from = "control_node/initialise" to = "matlab/initialise"/>
        <remap from = "control_node/planning_complete" to = "planning_node/planning_complete"/>
        <remap from = "control_node/update" to = "extended_kalman_filter_node/update"/>
        <remap from = "control_node/tof_range" to = "front_center_tof_driver_node/range"/>
        <remap from = "control_node/controlled_trajectory" to = "planning_node/controlled_trajectory"/>
        <remap from = "control_node/wheels_cmd" to = "wheels_driver_node/wheels_cmd"/>
        <node name = "$(arg node_name)" pkg = "$(arg pkg_name)" type = "$(arg node_name)" output = "screen">
            <param name = "veh" value = "$(arg veh)"/>
            <rosparam command = "load" file = "$(find motion_planner)/config/$(arg node_name)/$(arg parameter_file_name).yaml"/>
        </node>
    </group>
</launch>
